src/Workspaces/Core/Portable/Rename/RenameLocation.cs

```diff
@@ -69,5 +64,118 @@ public override int GetHashCode()
+                        // renamed.
+                        return false;
+
+                    case CandidateReason.NotAnEvent:
+                    case CandidateReason.NotAWithEventsMember:
+                        // it's unlikely that someone would be referencing a non-event in an 
+                        // event context.  Likely this location should be included.
+                        return false;
+
+                    case CandidateReason.NotAnAttributeType:
+                        // It's feasible that someon was referencing some type in an attribute
+                        // location before making that type itself descend from System.Attribute.
+                        // Still allow this type to be renamed.
+                        return true;
+
+                    case CandidateReason.WrongArity:
+                        // Someone may have provided the wrong number of type arguments to
+                        // a type/method when calling it. We should still allow the reference
+                        // to be updated.
+                        return true;
+
+                    case CandidateReason.NotCreatable:
+                        // Can happen when someone tries to do something like 'new' an 
+                        // abstract type.  We still want to allow renaming this location.
+                        return true;
+
+                    case CandidateReason.NotReferencable:
+                        // Happens when the user does something like directly accessing
+                        // the accessor of a normal property.  In this case, we do still
+                        // want to allow the rename to happen.
+                        return true;
+
+                    case CandidateReason.Inaccessible:
+                        // Can trivially occur in code that is in an initially broken state
+                        // where inaccessible members are being referenced.  We still want
+                        // to update these references.
+                        return true;
+
+                    case CandidateReason.NotAValue:
+                    case CandidateReason.NotAVariable:
+                        // Happens with code like "NS = 1".  If "NS" is binding now to a 
+                        // namespace, then it's likely something has gone very wrong (similar to
+                        // NotATypeOrNamespace), and we shouldn't update this reference.
+                        return false;
+
+                    case CandidateReason.NotInvocable:
+                        // Happens when something like a variable is being invoked, but the variable
+                        // isn't a delegate type.  This may be because the user intends to give 
+                        // this value a delegate type, but hasn't done so yet.  We should still allow
+                        // renaming this reference for now.
+                        return false;
```

Comment:
return true;

Also, there should probably be tests that cover all of these scenarios.